### PR TITLE
Add `datadog.leaderElectionResource` to select which resource lock to use in leader election

### PR DESCRIPTION
**Pull request by AliDatadog**
#### What this PR does / why we need it:
This PR adds `datadog.leaderElectionResource` parameter to select which resouce lock to use in leader election.

We added `leases` support in agent 7.47. Leases were added in Kubernetes 1.14. Since we still support older versions, we can not drop support of configmaps for leader election. Thus we need this parameter to enable/disable it depending on the agent and kubernetes version.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`
- [ ] For Datadog Operator chart or value changes update the test baselines (run: `make update-test-baselines`)

### CODE REVIEW

File: charts/datadog/CHANGELOG.md [hunk 1]
```diff
@@ -71,6 +75,7 @@
 ## 3.34.1
 
 * Fix `clusterAgent.admissionController.webhookName` RBAC to avoid restricting `create` by resource name.
+>>>>>>> main
```
Comment (celenechang) [inline charts/datadog/CHANGELOG.md:78]:
```suggestion
```

Comment (AliDatadog) [reply charts/datadog/CHANGELOG.md]: Thank you ! I missed this one

File: charts/datadog/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Datadog changelog
 
+## 3.39.0
+
+* Add a new parameter `datadog.leaderElectionResource` to select which resource lock to use in the leader election. Can be `leases(s)` in agent 7.47+, `configmap(s)` or empty for auto detection.
```
Comment (celenechang) [inline charts/datadog/CHANGELOG.md:5]:
```suggestion
* Add a new parameter `datadog.leaderElectionResource` to select which resource lock to use in the leader election. Can be `leases(s)` in agent 7.47+, `configmap(s)`, or empty for auto detection.
```

File: charts/datadog/values.yaml [hunk 3]
```diff
@@ -386,6 +386,13 @@ datadog:
   # datadog.leaderLeaseDuration -- Set the lease time for leader election in second
   leaderLeaseDuration:  # 60
 
+  # datadog.leaderElectionResource -- Selects the default resource to use for leader election.
+  # Can be:
+  # * "lease" / "leases". Only supported in agent 7.47+
+  # * "configmap" / "confimaps".
+  # "" to automatically detect which one to use.
```
Comment (celenechang) [inline charts/datadog/values.yaml:393]: sorry one more nit, could we also put a `*` here?
```suggestion
  # * "" to automatically detect which one to use.
```

**Review by celenechang — APPROVED** [review]

**Pull request merged**